File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    
    patterns = [
        [1, 2, 3, 4, 5],
        [2, 1, 2, 3, 2, 4, 2, 5],
        [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    ]
    
    scores = [0] * 3
    for i, ans in enumerate(answers):
        for j, ptn in enumerate(patterns):
            if ans == ptn[i % len(ptn)]:
                scores[j] += 1
                
    max_score = max(scores)
    
    highest_scores = []
    for i, score in enumerate(scores):
        if score == max_score:
            highest_scores.append(i + 1)
    
    return highest_scores

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;

class Solution {
    private static final int[][] PATTERNS = {
        {1, 2, 3, 4, 5},
        {2, 1, 2, 3, 2, 4, 2, 5},
        {3, 3, 1, 1, 2, 2, 4, 4, 5, 5}
    };

    public int[] solution(int[] answers) {
        int[] scores = new int[3];
        
        // 채점
        for (int i = 0; i < answers.length; i++) {
            int answer = answers[i];
            for (int j = 0; j < PATTERNS.length; j++) {
                if (answer == PATTERNS[j][i % PATTERNS[j].length]) {
                    scores[j]++;
                }
            }
        }

        // 최대 점수 계산
        int maxScore = Math.max(scores[0], Math.max(scores[1], scores[2]));

        // 결과
        List<Integer> result = new ArrayList<>();
        for (int i = 0; i < scores.length; i++) {
            if (scores[i] == maxScore) {
                result.add(i + 1);
            }
        }

        return result.stream()
                     .sorted()
                     .mapToInt(i -> i)
                     .toArray();
    }
}